react-router/lib/IndexRoute.d.ts

```diff
@@ -15,8 +15,8 @@ type ComponentsCallback = (err: any, components: RouteComponents) => any;
 export interface IndexRouteProps {
     component?: RouteComponent;
     components?: RouteComponents;
-    getComponent?(nextState: RouterState, callback: ComponentCallback): undefined;
```

Comment:
Needing the property syntax was a bug in the lint rule. Should be fixed by #14818